test/test-lib/github-mock.ts

```diff
@@ -0,0 +1,52 @@
+import * as Octokit from '@octokit/rest';
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import * as mockery from 'mockery';
+
+import getContent = require('../fixtures/github/getContent.json');
+
+class GHMock {
+	repos = {
+		getContents(
+			params?: Octokit.RequestOptions & Octokit.ReposGetContentsParams,
+		): Promise<Octokit.Response<Octokit.ReposGetContentsResponse>> {
+			if (!params) {
+				return Promise.reject(
+					new Error(`gh mock: No parameters provided for get content call.`),
+				);
+			}
+			const contentKey = `${params.owner}/${params.repo}/${params.path}`;
+
+			if (!_.has(getContent, contentKey)) {
+				return Promise.reject(
+					new Error(
+						`gh mock: getContents could not find a mock for ${contentKey}`,
+					),
+				);
+			}
+
+			const response: Octokit.Response<Octokit.ReposGetContentsResponse> = {
+				data: getContent[contentKey as keyof typeof getContent],
+				status: 200,
+				headers: {
+					date: 'Tue, 08 Oct 2019 14:53:13 GMT',
+					'x-ratelimit-limit': '60',
+					'x-ratelimit-remaining': '59',
+					'x-ratelimit-reset': '1570549993',
+					'x-Octokit-request-id': '123345',
+					'x-Octokit-media-type': 'github.v3; format=json',
+					link: '',
+					'last-modified': 'Wed, 02 Oct 2019 12:09:34 GMT',
+					etag: '63b1e55e670df084b122deba00768fe52ad424b7',
+					status: '',
+				},
+			} as Octokit.Response<Octokit.ReposGetContentsResponse>;
+			return Promise.resolve(response);
```

Comment:
If we are going to drop the bluebird import, then how about changing this to just `return response` and marking the method as `async`?